File: tools/improve/tests/original/maudec/cpp/truth-table.cpp



bool _and(bool a1, bool a2) {
	if ((a1 == true) && (a2 == true)) {
		return true;
	}
	if ((a1 == true) && (a2 == false)) {
		return false;
	}
	if ((a1 == false) && (a2 == true)) {
		return false;
	}
	if ((a1 == false) && (a2 == false)) {
		return false;
	}
}

bool andc(bool a1, bool a2) {
	if ((a1 == true) && (a2 == true)) {
		return true;
	}
	if ((a1 == false) && (a2 == false)) {
		return false;
	}
	if ((a1 == true) && (a2 == false)) {
		return false;
	}
	if ((a1 == false) && (a2 == true)) {
		return false;
	}
}

File: tools/improve/tests/original/maudec/py/truth-table.py


def and_(a1: bool, a2: bool) -> bool:
	match (a1, a2):
		case (True, True):
			return True

		case (True, False):
			return False

		case (False, True):
			return False

		case (False, False):
			return False



def andc(a1: bool, a2: bool) -> bool:
	match (a1, a2):
		case (True, True):
			return True

		case (False, False):
			return False

		case (True, False):
			return False

		case (False, True):
			return False



